test/SequenceEntity/SequenceEntity.js

```diff
@@ -13,14 +13,14 @@ describe('SequenceEntity', () => {
 			entityFactory(SequenceEntity, '/sequenceRoot', 'whatever', (entity) => {
 				entity.onSubSequencesChange((subSequence) => {
 					subSequence.onSequencedActivityChange((activity) => {
-						orgHref[activity.self()] = activity.organizationHrefs().pop();
+						orgHref[activity.index()] = activity.organizationHrefs().pop();
 					});
 				});
 			}, sirenSequenceRoot);
 
 			setTimeout(() => {
-				expect(orgHref['/activity/1']).to.equal('/organization/1');
-				expect(orgHref['/activity/2']).to.equal('/organization/2');
+				expect(orgHref[0]).to.equal('/organization/1');
+				expect(orgHref[1]).to.equal('/organization/2');
 				done();
 			}, 50);
```

Comment:
Is this because the parsing/loading occurs asynchronously? It would be nice to not rely on a timer in any tests. If the timer is too long, it unnecessarily slows down the tests. If the timer is too long, then the test is flaky. Is their some kind of "finished" event we could listen for instead to know that asynchronous work is complete?

Maybe something to consider for a future PR. I'm not suggesting fixing this now.

Reply:
Yes, I would love that. I don't know how though :(

Reply:
In other repos I've wrapped the callback in a promise and resolved after the expected number of invocations. It works alright if you pair it with async/await

Reply:
Jeff proposed a solution to make the entity factory to become an async so we can then wait on these results. I believe this change is going to be pretty good.

What repos did you do your method so I can look at it?

Reply:
Can you resolve async things multiple times? AFAIK awaitable things can only resolve once, which messes with the pattern we've set up for a listener

Reply:
Yes, so Jeff proposed change won't mess up our pattern. From what I gather. He going to put up a PR so we can review and see it.

Reply:
cool, looking forward to it